src/Aspire.Dashboard/Otlp/Model/OtlpApplicationView.cs

```diff
@@ -0,0 +1,58 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using Aspire.Dashboard.Otlp.Storage;
+using Google.Protobuf.Collections;
+using OpenTelemetry.Proto.Common.V1;
+
+namespace Aspire.Dashboard.Otlp.Model;
+
+[DebuggerDisplay("Application = {Application}, Properties = {Properties.Count}")]
+public class OtlpApplicationView
+{
+    public ApplicationKey ApplicationKey => Application.ApplicationKey;
+    public OtlpApplication Application { get; }
+    public KeyValuePair<string, string>[] Properties { get; }
+
+    public OtlpApplicationView(OtlpApplication application, RepeatedField<KeyValue> attributes)
+    {
+        Application = application;
+
+        var properties = new List<KeyValuePair<string, string>>();
```

Comment:
How often would an instance have no properties? If that weren't rare, we could save this allocation in such cases.

Reply:
There will always be properties, but it might just be the two known properties. I added optimization for that case.